src/Http/WebUtilities/src/FormPipeReader.cs

```diff
@@ -96,11 +96,12 @@ public FormPipeReader(PipeReader pipeReader, Encoding encoding)
 
                 if (readResult.IsCompleted)
                 {
+                    _pipeReader.AdvanceTo(buffer.End);
```

Comment:
Hmm I wonder if we should be calling Complete I the finally

Reply:
Any reason we shouldn't rely on the server auto-completing the body PipeReader?

Reply:
I don’t think that’s problematic but this logic did drain the pipe so why hold onto any buffers if we’re done?